metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py

```diff
@@ -109,19 +109,31 @@ def get_workunits(
             f"Upstream lineage detected for {self.report.num_tables_with_upstreams} tables.",
         )
 
-        yield from self.get_table_external_upstream_workunits()
+        if self._external_lineage_map:  # Some external lineage is yet to be emitted
+            yield from self.get_table_external_upstream_workunits(discovered_tables)
+            logger.info(
+                f"Only upstream external lineage detected for {self.report.num_tables_with_external_upstreams_only} tables.",
+            )
```

Comment:
This log statement makes more sense to me at the end of `get_table_external_upstream_workunits`, since that's the method that aggregates `self.report.num_tables_with_external_upstreams_only`

Reply:
done